### ISSUE TITLE
Code generator fails with ArrayIndexOutOfBoundsException for Index..Include statements

### ISSUE THREAD

**Issue by arvindth** [jOOQ/jOOQ#10466]
### Expected behavior and actual behavior:

When the code generator encounters an index with include statement, it throws an `ArrayIndexOutOfBoundsException`.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

Create the schema below in a postgres db, and then point to it with the codegen config further below:

```
DROP SCHEMA IF EXISTS mcve CASCADE;

CREATE SCHEMA mcve;

CREATE TABLE mcve.test (
  id SERIAL PRIMARY KEY,
  value INT,
  column_a text,
  column_b text,
  deleted boolean DEFAULT false NOT NULL
);

CREATE INDEX IF NOT EXISTS test_idx ON mcve.test (column_a, column_b) INCLUDE (deleted);
```

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration xmlns="http://www.jooq.org/xsd/jooq-codegen-3.13.0.xsd">
    <jdbc>
        <driver>org.postgresql.Driver</driver>
        <url>jdbc:postgresql://localhost/postgres</url>
        <user>postgres</user>
        <password></password>
    </jdbc>

    <generator>
        <database>
            <name>org.jooq.meta.postgres.PostgresDatabase</name>
            <includes>.*</includes>
            <inputSchema>mcve</inputSchema>
        </database>

        <generate>
          <generatedAnnotation>false</generatedAnnotation>
        </generate>

        <target>
            <packageName>org.jooq.mcve</packageName>
            <directory>target/generated-sources/jooq-mcve</directory>
        </target>
    </generator>
</configuration>
```
gives the following error:
```
INFO: Embeddables fetched      : 0 (0 included, 0 excluded)
Aug 03, 2020 9:00:55 PM org.jooq.tools.JooqLogger error
SEVERE: Error while generating table mcve.test
java.lang.RuntimeException: java.lang.ArrayIndexOutOfBoundsException: 2
	at org.jooq.meta.AbstractDatabase.onError(AbstractDatabase.java:2557)
	at org.jooq.meta.AbstractDatabase.getIndexes(AbstractDatabase.java:2034)
	at org.jooq.meta.AbstractDatabase.getIndexes(AbstractDatabase.java:2068)
	at org.jooq.meta.AbstractTableDefinition.getIndexes(AbstractTableDefinition.java:94)
	at org.jooq.codegen.JavaGenerator.generateTable(JavaGenerator.java:4113)
	at org.jooq.codegen.JavaGenerator.generateTable(JavaGenerator.java:3874)
	at org.jooq.codegen.JavaGenerator.generateTables(JavaGenerator.java:3861)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:519)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:473)
	at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:394)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:845)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:222)
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:217)
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:189)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 2
	at org.jooq.meta.postgres.PostgresDatabase$1.<init>(PostgresDatabase.java:234)
	at org.jooq.meta.postgres.PostgresDatabase.getIndexes0(PostgresDatabase.java:225)
	at org.jooq.meta.AbstractDatabase$14.run(AbstractDatabase.java:2037)
	at org.jooq.meta.AbstractDatabase.onError(AbstractDatabase.java:2547)
	... 13 more

```
Curiously, I haven't been able to reproduce the `ArrayIndexOutOfBoundsException` using the mcve though. It gives me a NullPointerException, which I haven't figured out how to debug yet.

### Versions:

- jOOQ: 3.13.4
- Java:1.8.0_212
- Database (include vendor): postgresql 
- OS: mac
- JDBC Driver (include name if inofficial driver): org.postgresql:postgresql:42.2.14

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'll try to reproduce this soon

**Comment by lukaseder** [comment]
I can reproduce this with:

```sql
CREATE TABLE t_10466 (
  a INT,
  b INT,
  c INT
);

CREATE INDEX i_10466 ON t_10466 (a) INCLUDE (b, c);
```

Will fix right away. Note that while we support `DSL.createIndex(...).on(...).include(...)` as a statement, we currently don't support generating this information in the code generator, neither for PostgreSQL nor SQL Server: https://github.com/jOOQ/jOOQ/issues/10469

**Comment by lukaseder** [comment]
Thanks again for your report.

Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.5 (#10470)
